contrib/google.golang.org/grpc/option.go

```diff
@@ -58,3 +58,45 @@ func NoDebugStack() InterceptorOption {
 		cfg.noDebugStack = true
 	}
 }
+
+type statsHandlerConfig struct {
+	serviceName  string
+	noDebugStack bool
+}
+
+func newStatsHandlerConfig() *statsHandlerConfig {
+	return &statsHandlerConfig{}
+}
+
+func (cfg *statsHandlerConfig) serverServiceName() string {
+	if cfg.serviceName == "" {
+		return "grpc.server"
+	}
+	return cfg.serviceName
+}
+
+func (cfg *statsHandlerConfig) clientServiceName() string {
+	if cfg.serviceName == "" {
+		return "grpc.client"
+	}
+	return cfg.serviceName
+}
+
+// StatsHandlerOption represents an option that can be passed
+// to the grpc client and server stats handlers.
+type StatsHandlerOption func(*statsHandlerConfig)
+
+// StatsHandlerOptionWithServiceName sets the given service name.
+func StatsHandlerOptionWithServiceName(name string) StatsHandlerOption {
```

Comment:
This naming is a bit nasty (aka long). I wonder if we can get away with using `InterceptorOption`. We could just ignore the ones that don't apply. What do you think?

Reply:
I agree with you, but I'm concerned about the name "InterceptorOption". What do you think about this change which uses [type alias](https://golang.org/ref/spec#Alias_declarations) for backward compatibility? https://github.com/zoncoen/dd-trace-go/commit/3d0c7eb9cbb066ee6398a3ed69675f3e60cd2c46#diff-d173ebd2cbc16a7769546ff9a6ee19caR24

Reply:
I wouldn't necessarily be against it but unfortunately we can't do that because it would be a breaking change (it's changing the function signatures).

> I agree with you, but I'm concerned about the name "InterceptorOption"

Fair enough, it was mostly just a question not something I feel strongly about. Let's leave it as it is, it is much nicer and better contained.

I would like however if we could make the names of the options shorter, but I don't have any great ideas or suggestions:

* `StatsOptServiceName`
* `StatsOptNoDebugStack`

How would these feel?

The current way is nice and it's definitely how it would work in most cases from Go's best practices, but I think this is one of those rare exceptions where it's a bit too long.

Reply:
I think it is not breaking change because `InterceptorOption` is declared as type alias. `type InterceptorOption = Option` means `InterceptorOption` and `Option` denote the same type. For example, the following code works without modification.

```go
package main

import (
	"google.golang.org/grpc"
	ddgrpc "gopkg.in/DataDog/dd-trace-go.v1/contrib/google.golang.org/grpc"
)

func main() {
	interceptor := createInterceptor(ddgrpc.WithServiceName("example"))

	// ... use interceptor
}

// ddgrpc.InterceptorOption works as ddgrpc.Option.
func createInterceptor(opts ...ddgrpc.InterceptorOption) grpc.UnaryClientInterceptor {
	return ddgrpc.UnaryClientInterceptor(opts...)
}

```

Reply:
Ah yes, of course, you're right! But I'd prefer to keep them specific, as they are. I feel like it would be a misuse of what type aliases are [meant to be used for](https://go.googlesource.com/proposal/+/master/design/18130-type-alias.md#background). It'd still lean in the favor of simply shortening.

Reply:
If the change has another problem, your suggestion `StasOptServiceName` is better I think.

Reply:
Then again I'm not strongly against your suggestion either. I'll let you decide between these two options and I'll be happy to approve both the "type alias" version and the shortening one. I'll leave it up to you 😄

Reply:
I changed the option names. PTAL 🙏